app/components/utils/government-areas/edit-government-areas-modal.js

```diff
@@ -22,12 +22,29 @@ export default class EditGovernmentAreasModal extends Component {
   @task
   *loadGovernmentAreas() {
     this.governmentFields = yield this.store.query('concept', {
-      'filter[top-concept-schemes][:uri:]':
-        CONSTANTS.CONCEPT_SCHEMES.BELEIDSVELD,
-      'filter[:has-no:end-date]': true,
+      filter: {
+        'top-concept-schemes': {
+          ':uri:': CONSTANTS.CONCEPT_SCHEMES.BELEIDSVELD,
+        },
+        ':lte:start-date': this.args.datetimeForGovernmentAreas.toISOString(),
+        ':gte:end-date': this.args.datetimeForGovernmentAreas.toISOString(),
+      },
       include: 'broader,narrower',
       'page[size]': 100,
     });
+    if (!this.governmentFields.length) {
```

Comment:
Is deze conditie correct? Kan de resultatenset geen mix bevatten van government-fields met en zonder einddatum?  (itt tot de mandatarissen selector waar we er wel vanuit mogen gaan dat ze allemaal samen vervangen worden)

Ben me aan het afvragen of we niet beter sowieso alle beleidsvelden ophalen uit de backend en de filtering op dates in de frontend doen. Door het toevoegen van de start-date en end-date filters aan de request gaan we nu veel meer cache misses hebben. Er zijn voorheen al eens opmerkingen geweest dat het dan te lang duurt.

Reply:
> Kan de resultatenset geen mix bevatten van government-fields met en zonder einddatum?

Normaal gezien niet :thinking:, want de beleidsvelden worden ook allemaal te samen vervangen. We hebben dan wel heel veel "duplicate" beleidsvelden.

Alles ophalen en in de frontend filteren is waarschijnlijk zelfs beter, want deze queries zijn inderdaad nogal traag.

Reply:
Note: alle beleidsvelden ophalen duurt (bij mij lokaal tenminste) ~1 minuut. Dat is op zich geen probleem op productie, de call kan toegevoegd worden aan de cache-warmup-service. Maar is misschien toch wel ambetant voor developers (want ik denk dat de cache-warmup niet by default draait in dev environment).

Reply:
Als de call in de application route zit, dan liefst mét includes. Anders moeten we ~700 calls doen bij het filteren in de edit-areas-modal.

Als we het weer in de edit-government-areas-modal component zit hoeven we geen extra calls te doen. Daar hoeven we enkel te filteren op start en einddatum, properties dus, waardoor er geen extra calls nodig zijn.

De call gaat nu weer naar de edit-government-areas-modal. Het verschil tussen met en zonder includes is heel klein. Mét: ~57s. Zonder: ~54s. De include is hier wel niet meer nodig.